node/src/accountant/payment_adjuster/non_unit_tests/mod.rs

```diff
@@ -292,140 +251,240 @@ fn make_payable_account(
-)> {
-    let payables = make_payables_according_to_thresholds_setup(
-        gn,
-        &thresholds_to_be_used,
-        accounts_count,
-        now,
-    );
+) -> Option<(u128, Vec<QualifiedPayableAccount>, AppliedThresholds)> {
+    let (payables, applied_thresholds) =
+        make_payables_according_to_thresholds_setup(gn, accounts_count, now);
 
-    let (qualified_payables, wallet_and_thresholds_pairs) =
-        try_make_qualified_payables_by_applied_thresholds(payables, &thresholds_to_be_used, now);
+    let qualified_payables =
+        try_make_qualified_payables_by_applied_thresholds(payables, &applied_thresholds, now);
 
-    let balance_average = {
-        let sum: u128 = sum_as(&qualified_payables, |account| account.balance_minor());
-        sum / accounts_count as u128
-    };
-    let cw_service_fee_balance_minor = {
-        let multiplier = 1000;
-        let max_pieces = accounts_count * multiplier;
-        let number_of_pieces = generate_usize(gn, max_pieces - 2) as u128 + 2;
-        balance_average / multiplier as u128 * number_of_pieces
-    };
-    let required_service_fee_total: u128 =
-        sum_as(&qualified_payables, |account| account.balance_minor());
+    let cw_service_fee_balance_minor =
+        pick_appropriate_cw_service_fee_balance(gn, &qualified_payables, accounts_count);
+
+    let required_service_fee_total: u128 = sum_as(&qualified_payables, |account| {
+        account.initial_balance_minor()
+    });
     if required_service_fee_total <= cw_service_fee_balance_minor {
         None
     } else {
         Some((
             cw_service_fee_balance_minor,
             qualified_payables,
-            wallet_and_thresholds_pairs,
+            applied_thresholds,
         ))
     }
 }
 
+fn pick_appropriate_cw_service_fee_balance(
+    gn: &mut ThreadRng,
+    qualified_payables: &[QualifiedPayableAccount],
+    accounts_count: usize,
+) -> u128 {
+    // Value picked empirically
+    const COEFFICIENT: usize = 1000;
+    let balance_average = sum_as(qualified_payables, |account| {
+        account.initial_balance_minor()
+    }) / accounts_count as u128;
+    let max_pieces = accounts_count * COEFFICIENT;
+    let number_of_pieces = generate_usize(gn, max_pieces - 2) as u128 + 2;
+    balance_average / COEFFICIENT as u128 * number_of_pieces
+}
+
 fn make_payables_according_to_thresholds_setup(
     gn: &mut ThreadRng,
-    thresholds_to_be_used: &AppliedThresholds,
     accounts_count: usize,
     now: SystemTime,
-) -> Vec<PayableAccount> {
-    match thresholds_to_be_used {
+) -> (Vec<PayableAccount>, AppliedThresholds) {
+    let wallets = prepare_account_wallets(accounts_count);
+
+    let nominated_thresholds = choose_thresholds(gn, &wallets);
+
+    let payables = match &nominated_thresholds {
         AppliedThresholds::Defaulted => make_payables_with_common_thresholds(
             gn,
+            wallets,
             &PRESERVED_TEST_PAYMENT_THRESHOLDS,
```

Comment:
There's another instance where you'd like to replace this constant with the `PaymentThresholds::default()`. In case you'd like to remove it from the other places, perhaps this comment can act as a reminder.